libbeat/common/mapstr.go

```diff
@@ -380,3 +389,43 @@ func (op putOperation) Do(key string, data MapStr) (interface{}, error) {
 	data[key] = op.Value
 	return existingValue, nil
 }
+
+type setOperation struct {
+	Value interface{}
+}
+
+func (op setOperation) Do(key string, data MapStr) (interface{}, error) {
```

Comment:
Then it would break with the [mapStrOperator interface](https://github.com/simitt/beats/blob/68fbee325f55615dcb402601adddb2290713e651/libbeat/common/mapstr.go#L351) which is used by all other methods. I just aligned using this interface to make it consistent.